Migration from XCTest to Swift Testing

Seb-Ltz opened the pull request:
# Migration from XCTest to Swift Testing

## :recycle: Current situation & Problem
As described in the issue #95, Swift Testing is already available, and existing packages can be migrated towards this new testing framework.

## :gear: Release Notes 
This PR updates all the 6 SpeziLLMTests (Parameter tests) of the `SpeziLLM` package from XCTest to Swift Testing

## :books: Documentation
Since this PR focuses on test refactoring, no documentation changes have been made.

## :white_check_mark: Testing
The test contents have remained unchanged.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

Seb-Ltz commented:
Also, since every test class only contained a single test, I didn't add any Swift Testing `@Suite`s. If you prefer to have those, let me know and I will adapt the PR.

philippzagar commented on Tests/SpeziLLMTests/LLMOpenAIParameterTests+OptionalTypes.swift:
```diff
@@ -80,94 +81,95 @@ final class LLMOpenAIParameterOptionalTypesTests: XCTestCase {
-        let parameterData = try XCTUnwrap(
-            JSONEncoder().encode(Parameters.shared)
+        let parameterData = try #require(
+            try JSONEncoder().encode(Parameters.shared)
```
I wonder if the inner `try` is really necessary? Don't know it off the top of my head but I think the outer `try` should apply there as well? Also applies to multiple pieces of the PR

Seb-Ltz replied:
This surprised me as well, but without this additional try inside of the `#require` macro I get this error:

> Call can throw, but it is not marked with 'try' and the error is not handled
> In expansion of macro 'require' here 

where xCode is highlighting the line inside the macro expansion: `$0.encode($1)`

If you have any idea to avoid that inner try, I would be really interested!

philippzagar replied:
Strange, thanks for the research here, let's just keep it as is! 🚀

philippzagar reviewed (commented):
Thanks for the great contribution and PR @Seb-Ltz, happy to see you work on SpeziLLM! 🚀 
Just had some thoughts regarding the overall Test suite structure and some minor in-line comments.
Don't worry about the failing CodeCov upload CI test, that's (sadly) expected for fork PRs.

Seb-Ltz commented on Tests/SpeziLLMTests/LLMOpenAIParameterTests+InvalidParameters.swift:
```diff
@@ -49,30 +50,33 @@ final class LLMOpenAIInvalidParametersTests: XCTestCase {
-        XCTAssertThrowsError(
-            try llmFunction.injectParameters(from: parameterData),
+        #expect(
+            throws: (any Error).self,
```
Good point! The `DecodingError.keyNotFound` error gets thrown here, however it seems that I can't `#expect` the `keyNotFound` case using SwiftTesting, but only the `DecodingError.self` type.

I've now kept the `#expect` macro with the `DecodingError.self` type. But if you prefer that we expect the respective `.keyNotFound` case, I can change the `#expect` macro to something like:
```swift
do {
    try llmFunction.injectParameters(from: parameterData)
    Issue.record("Expected DecodingError.keyNotFound but no error was thrown.")
} catch DecodingError.keyNotFound {
} catch {
    Issue.record("Unexpected error thrown: \(error.localizedDescription)")
}
```

philippzagar replied:
Thank you for the insights! I assumed that we wrap the decoding error within the `LLMOpenAIError` or something, but apparently we don't. Thanks!

Seb-Ltz commented on Tests/SpeziLLMTests/LLMOpenAIParameterTests+Array.swift:
```diff
@@ -61,62 +62,63 @@ final class LLMOpenAIParameterArrayTests: XCTestCase {
-        XCTAssert(!(try XCTUnwrap(llmFunction.parameterValueCollectors["doubleArrayParameter"])).isOptional)
-        XCTAssert(!(try XCTUnwrap(llmFunction.parameterValueCollectors["boolArrayParameter"])).isOptional)
-        XCTAssert(!(try XCTUnwrap(llmFunction.parameterValueCollectors["stringArrayParameter"])).isOptional)
+        #expect(!(try #require(llmFunction.parameterValueCollectors["intArrayParameter"])).isOptional)
+        #expect(!(try #require(llmFunction.parameterValueCollectors["doubleArrayParameter"])).isOptional)
+        #expect(!(try #require(llmFunction.parameterValueCollectors["boolArrayParameter"])).isOptional)
+        #expect(!(try #require(llmFunction.parameterValueCollectors["stringArrayParameter"])).isOptional)
```
Seems that swiftlint didn't throw any warnings 👍

Seb-Ltz commented:
Thank you very much for the review! I'm also glad to be working on SpeziLLM! 🚀
I've implemented the feedback from your comments, and also refactored the tests into one `@Suite`. If there's anything else, feel free to let me know

philippzagar reviewed (approved):
Thanks for incorporating the feedback @Seb-Ltz, looks great, ready to merge! 🚀

philippzagar commented:
@Seb-Ltz Let me know when you're ready to merge the PR! Sadly the CodeCov upload is failing (because of the fork) and I'll have to merge the PR with bypassing the merge requirements.

Seb-Ltz commented:
@philippzagar I would be ready to merge! 🚀
